src/controllers/AuthController.ts

```diff
@@ -62,3 +62,39 @@ export const login = async (
     next(exception);
   }
 };
+
+export const checkLogin = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const authHelper = new AuthHelper();
+
+    const { token } = req.cookies;
+    if (!token) throw new GenericError('Unauthorized', 401);
+
+    const { id }: any = await authHelper.verifyToken(token);
```

Comment:
It is better to add a type to the id variable, so it did not conflict with TS rules, I think it may be a string or a number, also for Ahmed, if an error occurred it would throw an error object, would it be caught in the catch block?

Reply:
I tried many types for this one, and they didn't work, I think it should be string | VerifyErrors but it didn't work.